lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -1559,8 +1559,9 @@ def plot_surface(self, X, Y, Z, *args, norm=None, vmin=None,
                 "Z contains NaN values. This may result in rendering "
                 "artifacts.")
 
-        # TODO: Support masked arrays
-        X, Y, Z = np.broadcast_arrays(X, Y, Z)
+        mask = (Z.mask if np.ma.isMaskedArray(Z)
+                else np.zeros_like(Z, dtype=bool))
```

Comment:
I think np ma has a helper for this

Reply:
Thanks for the pointer. I've replaced with `mask = np.ma.getmaskarray(Z)`.